llvmlite/tests/test_binding.py

```diff
@@ -1382,6 +1495,34 @@ def test_run(self):
         self.assertIn("%.4", orig_asm)
         self.assertNotIn("%.4", opt_asm)
 
+    def test_run_with_remarks(self):
+        mod = self.module(asm_inlineasm)
+        fn = mod.get_function("foo")
+        pm = self.pm(mod)
+        self.pmb().populate(pm)
+        mod.close()
+
+        pm.initialize()
+        remarkdesc, remarkfile = mkstemp()
+        try:
+            # Cannot find a trivial IR that will trigger a function pass to
+            # write a remark. So instead we write some junk into the file to
+            # observe that it has been overwritten.
```

Comment:
Something trivial might be the few lines of IR generated from 
```C
double licm(int num) {
    double acc = 0;
    for(int x = 0; x < num; x++){
        double invariant = 7;
        acc += invariant;
    }
    return (acc);
}
```
with `clang -S -emit-llvm -O0`. It ought to be possible to use the LICM pass through `pm.add_licm_pass()` to get some remarks, there should be some successes from moving the `load` of the `invariant` out.

Reply:
This was one of Siu's original test cases I left as is.